ApplyToBecomeInternal/ApplyToBecomeInternal/Pages/TaskList/PreviewHTBTemplate.cshtml.cs

```diff
@@ -0,0 +1,52 @@
+﻿using ApplyToBecome.Data.Models.KeyStagePerformance;
+using ApplyToBecome.Data.Services;
+using ApplyToBecomeInternal.Services;
+using ApplyToBecomeInternal.ViewModels;
+using Microsoft.AspNetCore.Mvc;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ApplyToBecomeInternal.Pages.TaskList
+{
+	public class PreviewHTBTemplateModel : BaseAcademyConversionProjectPageModel
+	{
+		private readonly KeyStagePerformanceService _keyStagePerformanceService;
+		private readonly ErrorService _errorService;
+
+		public PreviewHTBTemplateModel(KeyStagePerformanceService keyStagePerformanceService, IAcademyConversionProjectRepository repository, ErrorService errorService) : base(repository)
+		{
+			_keyStagePerformanceService = keyStagePerformanceService;
+			_errorService = errorService;
+		}
+
+		public TaskListViewModel TaskList { get; set; }
+		public bool ShowGenerateHtbTemplateError;
+		public string ErrorPage
+		{
+			set => TempData[nameof(ErrorPage)] = value;
+		}
+
+		public override async Task<IActionResult> OnGetAsync(int id)
+		{
+			await SetProject(id);
+
+			ShowGenerateHtbTemplateError = (bool)(TempData["ShowGenerateHtbTemplateError"] ?? false);
+			if (ShowGenerateHtbTemplateError)
+			{
+				// this sets the return location for the 'Confirm' button on the HeadTeacherBoardDate page
+				_errorService.AddError($"/task-list/{id}/confirm-school-trust-information-project-dates/head-teacher-board-date?return=%2FTaskList%2FPreviewHTBTemplate&fragment=head-teacher-board-date",
```

Comment:
You could do a similar change to what you did above by replacing `%2FTask....Template` with a variable that's a URL Encoded version of the `Links` page for the Preview page